django/thunderstore/account/test_service_account.py

```diff
@@ -0,0 +1,250 @@
+    assert (
+        form.errors["service_account"][0]
+        == "Must be identity owner to delete a service account"
+    )
+
+
+@pytest.mark.django_db
+def test_service_account_edit_nickname(service_account):
+    member = service_account.owner.members.first()
+    assert member.role == UploaderIdentityMemberRole.owner
+    form = EditServiceAccountForm(
+        member.user,
+        data={"service_account": service_account, "nickname": "New nickname"},
+    )
+    assert form.is_valid()
+    service_account = form.save()
+    assert service_account.user.first_name == "New nickname"
+    assert service_account.nickname == "New nickname"
+
+
+@pytest.mark.django_db
+def test_service_account_edit_nickname_too_long(service_account):
+    member = service_account.owner.members.first()
+    assert member.role == UploaderIdentityMemberRole.owner
+    form = EditServiceAccountForm(
+        member.user,
+        data={"service_account": service_account, "nickname": "x" * 1000},
+    )
+    assert form.is_valid() is False
+    assert len(form.errors["nickname"]) == 1
+    assert (
+        form.errors["nickname"][0]
+        == "Ensure this value has at most 32 characters (it has 1000)."
+    )
+
+
+@pytest.mark.django_db
+def test_service_account_edit_not_member(service_account):
+    user = UserFactory.create()
+    assert service_account.owner.members.filter(user=user).exists() is False
+    form = EditServiceAccountForm(
+        user,
+        data={"service_account": service_account, "nickname": "New nickname"},
+    )
+    assert form.is_valid() is False
+    assert len(form.errors["service_account"]) == 1
+    assert (
+        form.errors["service_account"][0]
+        == "Select a valid choice. That choice is not one of the available choices."
+    )
+
+
+@pytest.mark.django_db
+def test_service_account_edit_not_owner(service_account):
+    user = UserFactory.create()
+    UploaderIdentityMember.objects.create(
+        user=user,
+        identity=service_account.owner,
+        role=UploaderIdentityMemberRole.member,
+    )
+    form = EditServiceAccountForm(
+        user,
+        data={"service_account": service_account, "nickname": "New nickname"},
+    )
+    assert form.is_valid() is False
+    assert len(form.errors["service_account"]) == 1
+    assert (
+        form.errors["service_account"][0]
+        == "Must be identity owner to edit a service account"
+    )
+
+
+@pytest.mark.django_db
+def test_service_account_create_token(service_account):
+    member = service_account.owner.members.first()
+    assert member.role == UploaderIdentityMemberRole.owner
+    form = CreateTokenForm(
+        member.user,
+        data={"service_account": service_account},
+    )
+    assert form.is_valid()
+    token = form.save()
+    assert service_account.user == token.user
+
+
+def test_service_account_token_fixture(service_account_token):
+    assert service_account_token.user.service_account
```

Comment:
A bit confused how this works without a `@pytest.mark.django_db`, guessing fixtures just always have it enabled?

Reply:
Could be a race condition of some sort, e.g. the fixture being initialized while still in the scope of another test's @pytest.mark.django_db somehow. Could be good to add the marker on this test as well